common/network-common/src/main/java/org/apache/spark/network/util/TransportConf.java

```diff
@@ -324,6 +326,33 @@ public boolean separateChunkFetchRequest() {
     return conf.getInt("spark.shuffle.server.chunkFetchHandlerThreadsPercent", 0) > 0;
   }
 
+  /**
+   * Percentage of io.serverThreads used by netty to process FinalizeShuffleMerge. When the config
+   * `spark.shuffle.server.finalizeShuffleMergeThreadsPercent` is set, shuffle server will use a
+   * separate EventLoopGroup to process FinalizeShuffleMerge messages, which are I/O intensive and
+   * could take long time to process due to disk contentions. The number of threads used for handling
+   * finalizeShuffleMerge requests are percentage of io.serverThreads (if defined) else it is a
+   * percentage of 2 * #cores.
+   */
+  public int finalizeShuffleMergeHandlerThreads() {
+    if (!this.getModuleName().equalsIgnoreCase("shuffle")) {
+      return 0;
+    }
+    int finalizeShuffleMergeThreadsPercent =
+        Integer.parseInt(conf.get("spark.shuffle.server.finalizeShuffleMergeThreadsPercent"));
```

Comment:
Currently, this method should not be called when `spark.shuffle.server.finalizeShuffleMergeThreadsPercent` is not specified - but that is relying on current code flow in order for it to hold, which is brittle as project evolved.

Instead, handle explicitly this contract: 
a) Precondition check for `spark.shuffle.server.finalizeShuffleMergeThreadsPercent` to be present, or 
b) (more preferably) Handle the case when `spark.shuffle.server.finalizeShuffleMergeThreadsPercent` is missing.

Reply:
I think this is handled by `separateFinalizeShuffleMerge` we current have. 
1. When `spark.shuffle.server.finalizeShuffleMergeThreadsPercent` is missing, `separateFinalizeShuffleMerge` return false, so we will not call `finalizeShuffleMergeHandlerThreads` to create finalize handler.
2. Only when `spark.shuffle.server.finalizeShuffleMergeThreadsPercent` is larger than 0, we will create finalize handler.

Reply:
I covered this above :-)
What you are describing is how it currently works - but as project and code evolves, these assumption need not hold.
Let us make it either an explicit precondition (which will fail if/when the assumption is broken), or handle it such that we dont need to this assumption.

Reply:
I see your point. Add Preconditions. Please check. I also fixed all other suggestions. Thanks.